pkg/client/informers/informers_generated/generic.go

```diff
@@ -0,0 +1,249 @@
+
+package informers_generated
+
+import (
+	"fmt"
+	api "k8s.io/kubernetes/pkg/api"
+	api_v1 "k8s.io/kubernetes/pkg/api/v1"
+	apps "k8s.io/kubernetes/pkg/apis/apps"
+	v1beta1 "k8s.io/kubernetes/pkg/apis/apps/v1beta1"
+	autoscaling "k8s.io/kubernetes/pkg/apis/autoscaling"
+	v1 "k8s.io/kubernetes/pkg/apis/autoscaling/v1"
+	batch "k8s.io/kubernetes/pkg/apis/batch"
+	batch_v1 "k8s.io/kubernetes/pkg/apis/batch/v1"
+	v2alpha1 "k8s.io/kubernetes/pkg/apis/batch/v2alpha1"
+	certificates "k8s.io/kubernetes/pkg/apis/certificates"
+	v1alpha1 "k8s.io/kubernetes/pkg/apis/certificates/v1alpha1"
+	extensions "k8s.io/kubernetes/pkg/apis/extensions"
+	extensions_v1beta1 "k8s.io/kubernetes/pkg/apis/extensions/v1beta1"
+	policy "k8s.io/kubernetes/pkg/apis/policy"
+	policy_v1beta1 "k8s.io/kubernetes/pkg/apis/policy/v1beta1"
+	rbac "k8s.io/kubernetes/pkg/apis/rbac"
+	rbac_v1alpha1 "k8s.io/kubernetes/pkg/apis/rbac/v1alpha1"
+	storage "k8s.io/kubernetes/pkg/apis/storage"
+	storage_v1beta1 "k8s.io/kubernetes/pkg/apis/storage/v1beta1"
+	cache "k8s.io/kubernetes/pkg/client/cache"
+	schema "k8s.io/kubernetes/pkg/runtime/schema"
+)
+
+// GenericInformer is type of SharedIndexInformer which will locate and delegate to other
+// sharedInformers based on type
+type GenericInformer interface {
+	Informer() cache.SharedIndexInformer
+	Lister() cache.GenericLister
+}
+
+type genericInformer struct {
+	informer cache.SharedIndexInformer
+	resource schema.GroupResource
+}
+
+func (f *genericInformer) Informer() cache.SharedIndexInformer {
+	return f.informer
+}
+
+func (f *genericInformer) Lister() cache.GenericLister {
+	return cache.NewGenericLister(f.Informer().GetIndexer(), f.resource)
+}
+
+// ForResource gives generic access to a shared informer of the matching type
+// TODO extend this to unknown resources with a client pool
+func (f *sharedInformerFactory) ForResource(resource schema.GroupResource) (GenericInformer, error) {
```

Comment:
If you have a single factory with both internal and external types, you need a full `GroupVersionResource`